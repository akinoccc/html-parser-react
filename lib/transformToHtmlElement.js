/**
 * @name: AmsChan
 * @param {string} htmlstr
 * @param {object} data
 * @return {HTMLElement} dom[]
 */
export default (htmlstr, data = {}) => {
  const dataKeys = htmlstr.match(/\{.*\}/g) || [];
  dataKeys.forEach(target => {
    const key = target.replace(/[\{\}]/g, '');
    htmlstr = htmlstr.replace(target, data[key]);
  });
  const dom = document.createElement('div');
  dom.innerHTML = htmlstr;
  return dom.childNodes;
};
