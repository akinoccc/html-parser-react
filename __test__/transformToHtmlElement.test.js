import transformToHtmlElement from '../lib/transformToHtmlElement';
import htmlData from './testData/html.data';
import propsData from './testData/props.data';

describe('ReactParser', () => {
  it('parses html string', () => {
    expect(
      transformToHtmlElement(htmlData.renderData, propsData.data)
    ).toMatchSnapshot();
    expect(transformToHtmlElement(htmlData.singleHtmlDom)).toMatchSnapshot();
  });
});
