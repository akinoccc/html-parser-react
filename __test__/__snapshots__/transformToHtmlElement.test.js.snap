// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactParser parses html string 1`] = `
NodeList [
  <div>
    this is data1
    <span>
      this is data2
    </span>
  </div>,
]
`;

exports[`ReactParser parses html string 2`] = `
NodeList [
  <div>
    single test
  </div>,
]
`;
